src/intervals/set_operations.jl

```diff
@@ -56,18 +56,18 @@ function intersect{T}(a::Interval{T}, b::Interval{T})
     Interval(max(a.lo, b.lo), min(a.hi, b.hi))
 end
 # Specific promotion rule for intersect:
-intersect{T,S}(a::Interval{T}, b::Interval{S}) = intersect(promote(a,b)...)
+intersect{T,S}(a::Interval{T}, b::Interval{S}) = intersect(promote(a, b)...)
 
 
 ## Hull
 """
     hull(a, b)
 
-Returns the "convex hull" of the intervals `a` and `b`, considered as
-(extended) sets of real numbers. That is, the minimum set that contains
-all points in `a` and `b`.
+Returns the "interval hull" of the intervals `a` and `b`, considered as
+(extended) sets of real numbers, i.e. the smallest interval that contains
+all of `a` and `b`.
 """
-hull{T}(a::Interval{T}, b::Interval{T}) = Interval(min(a.lo, b.lo), max(a.hi, b.hi))
+hull{T}(a::Interval{T}, b::Interval{T}) = Interval{T}(min(a.lo, b.lo), max(a.hi, b.hi))
```

Comment:
I just notice that this may be too restrictive, so we need other methods when `eltype(a)` and `eltype(b)` are different. The following yields an error (also in master): `hull(Interval(1//2,3//1), Interval(1,2))`. The same applies to `union`.